libmisc/idmapping.c

```diff
@@ -189,21 +190,20 @@ void write_mapping(int proc_dir_fd, int ranges, const struct map_range *mappings
 
 	bufsize = ranges * ((ULONG_DIGITS + 1) * 3);
 	pos = buf = xmalloc(bufsize);
+	end = buf + bufsize;
 
 	/* Build the mapping command */
 	mapping = mappings;
 	for (idx = 0; idx < ranges; idx++, mapping++) {
 		/* Append this range to the string that will be written */
-		int written = snprintf(pos, bufsize - (pos - buf),
-			"%lu %lu %lu\n",
-			mapping->upper,
-			mapping->lower,
-			mapping->count);
-		if ((written <= 0) || ((size_t)written >= (bufsize - (pos - buf)))) {
-			fprintf(log_get_logfd(), _("%s: snprintf failed!\n"), log_get_progname());
-			exit(EXIT_FAILURE);
-		}
-		pos += written;
+		pos = shdw_stpeprintf(pos, end, "%lu %lu %lu\n",
+		                      mapping->upper,
+		                      mapping->lower,
+		                      mapping->count);
+	}
+	if (pos == end || pos == NULL) {
+		fprintf(log_get_logfd(), _("%s: snprintf failed!\n"), log_get_progname());
```

Comment:
s/snprintf/vsnprintf

Reply:
Or maybe shdw_stpeprintf? Just to forget about any implementation details hidden by the API

Reply:
Yeah, I wasn't sure what to log there.  Since most programmers know snprintf(3), and the documentation of shdw_[v]stpeprintf() defers to snprintf(3) in the ERRORS section, I chose to log snprintf() failure.  But I wasn't convinced by it.  Anyone else wants to suggest something? :)

Reply:
Fixed :)